source/memoizeWith.js

```diff
@@ -4,39 +4,47 @@ import _has from './internal/_has.js';
 
 
 /**
- * Creates a new function that, when invoked, caches the result of calling `fn`
- * for a given argument set and returns the result. Subsequent calls to the
- * memoized `fn` with the same argument set will not result in an additional
- * call to `fn`; instead, the cached result for that set of arguments will be
- * returned.
+ * Takes a string-returning function `sig` and a function `fn` and returns
+ * a new function that returns cached results for subsequent
+ * calls with the same arguments.
  *
- * Care must be taken when implementing key generation to avoid key collision,
+ * When the function is invoked, `sig` is applied to the same arguments
+ * and its result becomes the cache key. If the cache contains something
+ * under that key, the function simply returns it and does not invoke `fn` at all.
+ *
+ * Otherwise `fn` is applied to the same arguments and its return value
+ * is cached under that key and returned by the function.
+ *
+ * Care must be taken when implementing `sig` to avoid key collision,
  * or if tracking references, memory leaks and mutating arguments.
  *
  * @func
  * @memberOf R
  * @since v0.24.0
  * @category Function
  * @sig (*... -> String) -> (*... -> a) -> (*... -> a)
- * @param {Function} fn The function to generate the cache key.
+ * @param {Function} sig The function to generate the cache key.
```

Comment:
Would `kegGen` be a more explicit name?

Reply:
You mean `keyGen`? Sure that sounds good to me. Will change.

Reply:
Done.